openquake/hazardlib/gsim/sharma_2009.py

```diff
@@ -0,0 +1,232 @@
+    #: implementation.
+    non_verified = True
+
+    def get_mean_and_stddevs(self, sites, rup, dists, im_type, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for specification of input and result values.
+        """
+                
+        # extract dictionary of coefficients specific to required
+        # intensity measure type
+        C = self.COEFFS[im_type]
+        C.update(self.CONSTS)
+
+        log_mean = (self._compute_magnitude(rup, C) +
+                    self._compute_distance(dists, C) +
+                    self._get_site_amplification(sites, C) +
+                    self._get_mechanism(rup, C))
+
+        # Convert to g
+        mean = np.log((10.0 ** log_mean) / g)
+
+        stddevs = self._get_stddevs(C, stddev_types, 
+                                        len(sites.vs30))
+
+        # Since sigma is not present in equation (1) it is not 
+        # entirely clear what it represents. For now just pass the 
+        # raw value on through.       
+
+        return mean, stddevs
+
+    def _get_stddevs(self, C, stddev_types, num_sites):
+        """
+        Return total sigma as reported in Table 2, p. 1202.
+        """
+        stddevs = []
+        for stddev_type in stddev_types:
+            assert stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+            stddevs.append(C['sigma'] + np.zeros(num_sites))
+        return np.array(stddevs)
+
+    def _compute_magnitude(self, rup, C):
+        """
+        Compute first two terms of equation (1) on p. 1200:
+
+        ``b1 + b2 * M``
+        """
+        return C['b1'] + C['b2']*rup.mag
+
+    def _compute_distance(self, dists, C):
+        """
+        Compute third term of equation (1) on p. 1200:
+
+        ``b3 * log(sqrt(Rjb ** 2 + b4 ** 2))``
+        """
+        return C['b3']*np.log10(np.sqrt(dists.rjb**2. + C['b4']**2.))
+
+    def _get_site_amplification(self, sites, C):
+        """
+        Compute fourth term of equation (1) on p. 1200:
+
+        ``b5 * S``
+        """
+        S = self._get_site_type_dummy_variables(sites)
+        return C['b5']*S
+        
+    def _get_mechanism(self, rup, C):
+        """
+        Compute fifth term of equation (1) on p. 1200:
+
+        ``b6 * H``
+        """
+        H = self._get_fault_type_dummy_variables(rup)
+        return C['b6']*H
+        
+    def _get_site_type_dummy_variables(self, sites):
+        """
+        Binary rock/soil classification dummy variable, ``S`` based 
+        on sites.vs30. Theshhold is not explicitly given in Sharma 
+        et al. (2009) so a value of 760 m/s was chosen. This is 
+        consistent with 
+        :mod:`openquake.hazardlib.gsim.atkinson_boore_2003`, 
+        corresponds to NEHRP class A/B, and is close to the 
+        threshhold for Eurocode 8 Class 8 (800 m/s).
+        """
+        Sab = np.array(sites.vs30 > 760.)
```

Comment:
I would give an uppercase name to this constant. If it is the same as in `openquake.hazardlib.gsim.atkinson_boore_2003`, let's define it only once.